webrtc/lib/webrtc_endpoint.ex

```diff
@@ -374,24 +374,33 @@ defmodule Membrane.RTC.Engine.Endpoint.WebRTC do
 
     {:endpoint, endpoint_id} = ctx.name
 
-    Enum.each(new_outbound_tracks, fn track ->
-      case Engine.subscribe(state.rtc_engine, endpoint_id, track.id) do
-        :ok ->
-          :ok
-
-        {:error, :invalid_track_id} ->
-          Membrane.Logger.debug("""
-          Couldn't subscribe to the track: #{inspect(track.id)}. No such track.
-          It was probably removed before we restarted ICE. Ignoring.
-          """)
-
-        {:error, reason} ->
-          raise "Couldn't subscribe to the track: #{inspect(track.id)}. Reason: #{inspect(reason)}"
-      end
-    end)
+    {valid_tracks, invalid_tracks} =
+      Enum.split_with(new_outbound_tracks, fn track ->
+        case Engine.subscribe(state.rtc_engine, endpoint_id, track.id) do
+          :ok ->
+            true
+
+          {:error, :invalid_track_id} ->
+            Membrane.Logger.debug("""
+            Couldn't subscribe to the track: #{inspect(track.id)}. No such track.
+            It was probably removed before we restarted ICE.
+            Tracks from the offending origin (#{inspect(track.origin)}) will be renegotiated.
+            """)
+
+            false
+
+          {:error, reason} ->
+            raise "Couldn't subscribe to the track: #{inspect(track.id)}. Reason: #{inspect(reason)}"
+        end
+      end)
 
-    send_if_not_nil(state.display_manager, {:subscribe_tracks, ctx.name, new_outbound_tracks})
-    {[], state}
+    actions =
+      if Enum.empty?(invalid_tracks),
+        do: [],
+        else: build_track_removed_actions(invalid_tracks, ctx)
```

Comment:
As it always return a list I think it could be changed this way
```suggestion
    actions = build_track_removed_actions(invalid_tracks, ctx)
```